File: app/src/androidTest/java/com/github/sdpcoachme/BoredActivityTest.java
package com.github.sdpcoachme;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
import static androidx.test.espresso.matcher.ViewMatchers.withId;

import android.content.Intent;

import androidx.test.core.app.ActivityScenario;
import androidx.test.core.app.ApplicationProvider;

import org.junit.Test;

public class BoredActivityTest {
    @Test
    public void boredActivityContainsAllElements(){
        Intent intent = new Intent(ApplicationProvider.getApplicationContext(), BoredActivity.class);
        ActivityScenario<MainActivity> scenario = ActivityScenario.launch(intent);

        onView(withId(R.id.response)).check(matches(isDisplayed()));
        onView(withId(R.id.request)).check(matches(isDisplayed()));
        onView(withId(R.id.db)).check(matches(isDisplayed()));
        onView(withId(R.id.delete)).check(matches(isDisplayed()));

        scenario.close();
    }
}

File: app/src/androidTest/java/com/github/sdpcoachme/BoredActivityTest.kt
package com.github.sdpcoachme

import android.content.Intent
import androidx.test.core.app.ActivityScenario
import androidx.test.core.app.ApplicationProvider
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.matcher.ViewMatchers
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.ext.junit.runners.AndroidJUnit4
import dagger.hilt.android.testing.HiltAndroidRule
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class BoredActivityTest {
    @get:Rule
    val hiltRule = HiltAndroidRule(this)

    @Test
    fun boredActivityContainsAllElements() {
        val intent = Intent(ApplicationProvider.getApplicationContext(), BoredActivity::class.java)
        val scenario = ActivityScenario.launch<MainActivity>(intent)

        onView(withId(R.id.response)).check(matches(ViewMatchers.isDisplayed()))
        onView(withId(R.id.request)).check(matches(ViewMatchers.isDisplayed()))
        onView(withId(R.id.db)).check(matches(ViewMatchers.isDisplayed()))
        onView(withId(R.id.delete)).check(matches(ViewMatchers.isDisplayed()))

        scenario.close()
    }

//    @Test
//    fun fakeApiCalls() {
//        val intent = Intent(ApplicationProvider.getApplicationContext(), BoredActivity::class.java)
//        val scenario = ActivityScenario.launch<MainActivity>(intent)
//
//        onView(withId(R.id.delete)).perform(click())
//        onView(withId(R.id.response)).check(matches(withText("DB deleted")))
//
//        scenario.close()
//    }
}